### ISSUE TITLE
KineticEnergy treats mass in an unreadable manner

### ISSUE THREAD

**Issue by elad2718** [netket/netket#1562]
The following works as expected only when the `mass` argument it one over the mass of the particle. This is taken cared of in `_pack_arguments` so results are accidentally correct.
```python
    def _expect_kernel_single(
        self, logpsi: Callable, params: PyTree, x: Array, mass: Optional[PyTree]
    ):
```

**Comment by gcarleo** [comment]
it is a very ugly function, I agree, but it is correct, in the sense that it is a private function that users are not supposed to call directly. in the user-called functions masses are handled correctly

**Comment by gcarleo** [comment]
@gpescia if you have some time to fix the naming here, it would be great. This issue has come up already a few times and tends to confuse people who read the internals of the code!

**Comment by gcarleo** [comment]
fixed by #1566
